src/main/java/model/input/file/ExcelFile.java

```diff
@@ -72,19 +84,53 @@ private void addRows(Iterator<Row> rowIterator) {
 		}
 	}
 
-	private DataValue toDataValue(Cell cell) {
+	private DataValue toDataValue(Cell cell, Class<? extends DataValue> type) {
 		switch (cell.getCellType()) {
 			case Cell.CELL_TYPE_STRING:
 				return new StringValue(cell.getStringCellValue());
 			case Cell.CELL_TYPE_NUMERIC:
-				double cellValue = cell.getNumericCellValue();
-				return (cellValue % 1 == 0)
-						? new IntValue((int) cellValue) : new FloatValue((float) cellValue);
+				return parseNumValue(cell, type);
 			case Cell.CELL_TYPE_BLANK:
-				return new StringValue("");
-			default: throw new UnsupportedOperationException(
-					String.format("Cell type %s not supported", cell.getCellType())
-			);
+				return createNullValue(type);
+			default:
+				throw new UnsupportedOperationException(
+						String.format("Cell type %s not supported", cell.getCellType()));
+		}
+	}
+
+	private DataValue parseNumValue(Cell cell, Class<? extends DataValue> type) {
```

Comment:
ik kan het zelf nu niet zo makkelijk checken dus ik vraag het maar. Hou je rekening dat de numvalues voor excel datums anders is dan de numvalue van de calendar? de calendar gaat uit van de unix time stamp excel niet.